src/appshell/view/preferences/exportpreferencesmodel.cpp

```diff
@@ -0,0 +1,288 @@
+{
+    return EXPORT_PROCESS_MAPPING[exportConfiguration()->processType()].translated();
+}
+
+void ExportPreferencesModel::setCurrentProcess(const QString& newProcess)
+{
+    ExportProcessType type;
+    for (auto process : EXPORT_PROCESS_MAPPING) {
+        if (newProcess == process.second.translated()) {
+            type = process.first;
+        }
+    }
+
+    if (newProcess == currentProcess()) {
+        return;
+    }
+
+    exportConfiguration()->setProcess(type);
+}
+
+QVariantList ExportPreferencesModel::processList() const
+{
+    QVariantList result;
+    for (const auto& process : EXPORT_PROCESS_MAPPING) {
+        result << process.second.translated().toQString();
+    }
+
+    return result;
+}
+
+QString ExportPreferencesModel::filename() const
+{
+    return QString::fromStdString(exportConfiguration()->filename());
+}
+
+void ExportPreferencesModel::setFilename(const QString& filename)
+{
+    exportConfiguration()->setFilename(filename.toStdString());
+}
+
+QString ExportPreferencesModel::fileExtension() const
+{
+    return QString::fromStdString(exporter()->formatExtension(currentFormat().toStdString()));
+}
+
+QString ExportPreferencesModel::directoryPath() const
+{
+    return exportConfiguration()->directoryPath().toQString();
+}
+
+void ExportPreferencesModel::setDirectoryPath(const QString& path)
+{
+    if (path == directoryPath()) {
+        return;
+    }
+
+    exportConfiguration()->setDirectoryPath(path);
+}
+
+QString ExportPreferencesModel::currentFormat() const
+{
+    return QString::fromStdString(exportConfiguration()->currentFormat());
+}
+
+void ExportPreferencesModel::setCurrentFormat(const QString& format)
+{
+    if (format == currentFormat()) {
+        return;
+    }
+
+    exportConfiguration()->setCurrentFormat(format.toStdString());
+}
+
+QVariantList ExportPreferencesModel::formatsList() const
+{
+    QVariantList result;
+    for (const auto& format : exporter()->formatsList()) {
+        result << QString::fromStdString(format);
+    }
+
+    return result;
+}
+
+ExportChannelsPref::ExportChannels ExportPreferencesModel::exportChannels() const
+{
+    return exportConfiguration()->exportChannels();
+}
+
+void ExportPreferencesModel::setExportChannels(ExportChannelsPref::ExportChannels exportChannels)
+{
+    exportConfiguration()->setExportChannels(exportChannels);
+}
+
+int ExportPreferencesModel::maxExportChannels() const
+{
+    return exporter()->maxChannels();
+}
+
+QString ExportPreferencesModel::exportSampleRate() const
+{
+    auto currentSampleRate = exportConfiguration()->exportSampleRate();
+    for (const auto& rate : m_sampleRateMapping) {
+        if (currentSampleRate == rate.first) {
+            return rate.second;
+        }
+    }
+
+    return {};
+}
+
+QVariantList ExportPreferencesModel::exportSampleRateList()
+{
+    std::vector<int> sampleRateList = exporter()->sampleRateList();
+    if (sampleRateList.empty()) {
+        sampleRateList = DEFAULT_SAMPLE_RATE_LIST;
```

Comment:
If the exporter doesn’t support sampleRate (returns an empty list), then we can’t forcibly set a specific sampleRate. Am I right? 
(The model is supposed to only receive data from the service and display it - without adding any extra logic)